src/Element.js

```diff
@@ -0,0 +1,341 @@
+			return value
+		} else {
+			switch (typeof value) {
+				case 'number':
+				case 'string':
+					return text(value, index)
+				case 'function':
+				case 'object':
+					if (Utility.iterable(value)) {
+						if (index === Constant.key) {
+							return iterable(Constant.fragment, value.map(from), Utility.hash(index))
+						} else {
+							return iterable(Constant.iterable, value.map(from), Utility.hash(index))
+						}
+					} else {
+						return create(value, Constant.object)
+					}
+			}
+		}
+	}
+
+	return empty(index)
+}
+
+/**
+ * @param {*} value
+ * @return {number}
+ */
+export function identity (value) {
+	switch (typeof value) {
+		case "string":
+			return Constant.node
+		case 'function':
+			return Constant.component
+		case 'number':
+			return Constant.fragment
+		case 'object':
+			return Utility.thenable(value) ? Constant.thenable : Constant.portal
+	}
+}
+
+/**
+ * @param {*} value
+ * @return {boolean}
+ */
+export function properties (value) {
+	if (value) {
+		if (typeof value === 'object') {
+			switch (value.constructor) {
+				case Object:
+					return true
+				case Constant.element:
+					break
+				default:
+					return !Utility.iterable(value)
+			}
+		}
+	}
+
+	return false
+}
+
+/**
+ * @param {*} value
+ * @return {string}
+ */
+export function display (value) {
+	switch (typeof value) {
+		case 'function':
+			return display(value[Constant.displayName] || value.name)
+		case 'object':
+			if (valid(value)) {
+				return display(value.type)
+			}
+		default:
+			return value.toString()
+	}
+}
+
+/**
+ * @param {object} element
+ * @param {object?} value
+ * @return {object}
+ */
+export function defaults (element, value) {
+	if (value) {
+		if (typeof value === 'function') {
+			defaults(element, value.call(element.type, element.props))
+		} else {
+			for (var key in value) {
+				if (element.props[key] === undefined) {
+					element.props[key] = value[key]
+				}
+			}
+		}
+	}
+
+	return element
+}
+
+/**
+ * @param {*} element
+ * @return {boolean}
+ */
+export function valid (element) {
+	return element != null && element.constructor === Constant.element
+}
+
+/**
+ * @param {object} value
+ * @param {...*?}
+ * @return {object}
+ */
+export function clone (element) {
+	return defaults(create.apply(null, [element.type].concat([].slice.call(arguments, 1))), element.props)
+}
+
+/**
+ * @param {object} element
+ * @param {number} from
+ * @return {object}
+ */
+export function parent (element, from) {
+	if (element.uuid < Constant.portal) {
+		if (element.uuid !== -from) {
+			return parent(element.parent, from, element, to)
+		}
+	}
+
+	return element
+}
+
+/**
+ * @param {object} element
+ * @return {object}
+ */
+export function node (element) {
+	if (element.uuid < Constant.node) {
+		return node(element.children[Constant.node])
+	}
+
+	return element
+}
+
+/**
+ * @param {Array<*>} element
+ * @param {*} value
+ * @return {*}
+ */
+export function put (element, value) {
```

Comment:
push returns an index, this might need to go into Utility, but we only use it for Element thus why it's here. Maybe not the best naming